fix: use a group role to allow screen readers to navigate the hierarchy

domoritz opened the pull request.

domoritz commented on packages/vega-scenegraph/src/util/aria.js:
```diff
@@ -6,6 +6,7 @@ const ARIA_LABEL = 'aria-label';
 const ARIA_ROLE = 'role';
 const ARIA_ROLEDESCRIPTION = 'aria-roledescription';
 const DEFAULT_ROLE = 'graphics-symbol';
```
Should we rename this to `MARK_ROLE` or `INDIVIDUAL_MARK_ROLE`?

jheer replied:
Or maybe rename to `SYMBOL_ROLE`? Also which of these should be used for guides?

domoritz replied:
Guides should use some role that does not have children so that VoiceOver does not read `empty X` after reading the description.

domoritz commented:
Actually, there is a bigger issue here. We need to set aria-role on individual marks, not on the group that surrounds marks. 

Notice in this SVG (I manually changed `graphics-symbol` to `graphics-object`)

<img width="304" alt="Screen Shot 2020-04-23 at 09 13 41" src="https://user-images.githubusercontent.com/589034/80122682-bb019400-8542-11ea-9367-d598435388d6.png">

the description is set on the group mark but that means the individual children don't get the right role description.

jheer commented:
To clarify: for the screen reader to properly navigate the individual marks, do the mark svg elements needs to have a `role` property set? If so, you could set that using the `ariaRole` encoding channel, yes? (We could possibly automate the choice of `role` if `description` is defined.) Remember that by default we don't add any annotations to marks because that can cause massive bloat to both the scenegraph and output SVG. If we automate the `role` choice by default (assuming that is necessary) that will at least reduce bloat to the scenegraph objects.

domoritz commented:
It depends a bit on the screenreader (unfortunately it's like in times of IE6) but generally only `<text>` elements are read even without an explicit `aria-label`, `aria-role`, or `aria-roledescription` property. So if we want to not have VoiceOver read out an element, we should not set any property.

domoritz commented:
In my initial pull request, I decided to add aria roles and aria roledescriptions so that even when the developer does not provide any aria annotations, the screenreader shows something and the person accessing the chart with a screenreader knows that there is something. Vega-Lite will automatically generate a description for marks since it has the high-level semantic information. I filed https://github.com/vega/vega/issues/2494 as a follow up to automatically add annotations for individual marks in Vega.

What I like about your approach is that groups of marks are summarized into a single annotation (solving https://github.com/vega/vega/issues/2492). I think what we need to do is add the aria properties to individual marks when they are explicitly provided by the user.

I'm closing this pull request since it's not really addressing the issues I described in [my review](https://github.com/vega/vega/pull/2549#pullrequestreview-399267475).
